shared/lsp/connection.ts

```diff
@@ -24,12 +24,12 @@ export interface LSPConnection extends sourcegraph.Unsubscribable {
 
 export const webSocketTransport = ({
     serverUrl,
-    logger,
     cancellationToken,
+    logger = console,
```

Comment:
We need to be careful with using `console` directly, it should be wrapped at least in a `RedactedLogger`. Logs may contain the root URI (in errors, objects or whatever) which contains the access token, and we had people accidentally post those logs to our public issue tracker. I'd recommend against a default value

Reply:
Marked every use of console with TODO (will clean these up in a next commit).

Reply:
Sounds good, thanks. This rule may help to enable just in this repo (and disable with a comment in the place where it's used legitimately): https://eslint.org/docs/rules/no-console

Reply:
Updated.